api/enterprise-api-ref/two-factor-authentication.markdown

```diff
@@ -0,0 +1,239 @@
+---
+layout: default
+title: Two-factor authentication API
+published: true
+---
+The Two-factor authentication API enables users to add an extra layer of security to their accounts
+by requiring a TOTP (time-based one-time password) in addition to their primary credentials.
+
+Once two-factor authentication is enabled, requests to obtain an OAuth token, change a password, or
+disable two-factor authentication must include an additional header: Cf-2fa-Token: code.
+Basic authentication also requires this header each time it is used.
+
+Currently, only TOTP (Time-based One-Time Password) two-factor authentication
+is supported, providing users with a time-sensitive code for enhanced security.
+
+## Start two-factor authentication configuration
+
+First, you need to request two-factor authentication secrets to complete the configuration.
+The secret token should be added to a third-party authentication app such as Google Authenticator or Authy.
+A 6-digit code should then be obtained from the application to complete the action.
+
+**URI:** https://hub.cfengine.com/api/2fa/totp/configure
+
+**Method:** GET
+
+**Example request (curl):**
+
+```console
+curl --user <username>:<password> \
+  -X GET \
+  https://hub.cfengine.com/api/2fa/totp/configure
+```
+
+**Successful response example:**
+
+```
+HTTP 200 Ok
+{
+    "secret": "SYSZJJVQUIOG5ITA6HPH5PZWLZ3VUP52",
+    "2faUrl": "otpauth://totp/Mission%20Portal:admin?secret=SYSZJJVQUIOG5ITA6HPH5PZWLZ3VUP52&issuer=Mission%20Portal&algorithm=SHA1&digits=6&period=30",
+    "algorithm": "sha1",
+    "digits": 6,
```

Comment:
can we include a description of the fields?

digits?
period is... days?
holder is username?

Reply:
Good suggestion.

Reply:
added output section

Reply:
cool, thanks!